src/main/scala/activemq/cli/command/Commands.scala

```diff
@@ -139,38 +139,53 @@ abstract class Commands extends PrintStackTraceExecutionProcessor {
   }
 
   def withBroker(callback: (BrokerViewMBean, MBeanServerConnection) ⇒ String): String = {
+    var isConnected = false
+    var connectionMessage = ""
     ActiveMQCLI.broker match {
       case Some(matched) ⇒
-        val jmxConnector = JMXConnectorFactory.connect(
-          new JMXServiceURL(matched.jmxurl),
-          mapAsJavaMap(Map(CREDENTIALS → Array(matched.username, matched.password)))
-        )
-        try {
-          jmxConnector.connect
-          // Fuse ESB Enterprise 7.1.0 / ActiveMQ 5.9.0 use different ObjectNames
-          val brokerViewMBeans = List(Map("type" → "type", "brokerName" → "brokerName"), Map("type" → "Type", "brokerName" → "BrokerName"))
-            .par.map(properties ⇒
-              jmxConnector.getMBeanServerConnection.queryNames(
-                new ObjectName(s"org.apache.activemq:${properties.get("type").get}=Broker,${properties.get("brokerName").get}=${matched.jmxName.getOrElse("*")}"), //scalastyle:ignore
-                null //scalastyle:ignore
-              )).flatten
-            .par.map(objectName ⇒ MBeanServerInvocationHandler.newProxyInstance(
-              jmxConnector.getMBeanServerConnection, objectName, classOf[BrokerViewMBean], true
-            ))
-            .filter(!_.isSlave)
-
-          brokerViewMBeans.headOption match {
-            case Some(brokerViewMBean) ⇒ callback(brokerViewMBean, jmxConnector.getMBeanServerConnection())
-            case _                     ⇒ "Broker not found"
+        val jmxurls = matched.jmxurl.split(",")
+        jmxurls.foreach { url ⇒
+          if (!isConnected) {
+            val jmxConnector = JMXConnectorFactory.connect(
```

Comment:
If `JMXConnectorFactory.connect` throws an exception it breaks out of the foreach and subsequent jmxurl's are not tried.

I can verify this by running below commands:
`activemq-cli>start-embedded-broker`
`activemq-cli>connect --broker local`

with below broker config (first jmxurl contains an unknown host):

```
local {
    amqurl = "tcp://localhost:61616"
    jmxurl = "service:jmx:rmi:///jndi/rmi://UNKNOWN:1099/jmxrmi,service:jmx:rmi:///jndi/rmi://localhost:1099/jmxrmi"
    username = "admin"
    password = "admin"
    prompt-color = "light-blue"
  }
```

Reply:
Although I think there are more exceptions (other than ServiceUnavailableException) that qualify for the JMX 'failover' I have no problem merging this pr. Thanks for submitting!